src/lib/contact.ts

```diff
@@ -279,6 +279,40 @@ class Contact {
 		}
 	}
 
+	public async moveFileToBin(concernedFile: IPCFile, deletedAt: number | null, sharedFiles: IPCFile[]): Promise<ResponseType> {
+		try {
+			let fileFound = false;
+			this.contacts.forEach(async (contact) => {
+				const file = contact.files.find((f) => f.id === concernedFile.id);
+
+				if (file) {
+					file.deletedAt = deletedAt;
+					fileFound = true;
+					await this.publishAggregate();
+				}
+			});
+			if (!fileFound) {
+				const file = sharedFiles.find((f) => f.id === concernedFile.id);
+				if (!file) {
+					return { success: false, message: 'File not found' };
+				}
+				await post.Publish({
+					account: this.account!,
+					postType: 'InterPlanetaryCloud',
+					content: { file: { ...concernedFile, deletedAt }, tags: ['rename', concernedFile.id] },
```

Comment:
great to support shared content sent to the bin!

But this won't work, you also need to take care of this `rename` case is the `loadUpdates` method

Reply:
Done